pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/SortOperatorTest.java

```diff
@@ -472,6 +473,29 @@ public void shouldHandleMultipleCollationKeysWithNulls() {
     assertTrue(operator.nextBlock().isSuccess(), "expected EOS block to propagate");
   }
 
+  @Test
+  public void shouldPreservePrecision() {
+    // Given:
+    DataSchema schema = new DataSchema(new String[]{"sort"}, new DataSchema.ColumnDataType[]{LONG});
+    // Insert 3 large long values which will be the same if converted to double due to precision loss.
```

Comment:
"which will be the same" ==> "which will NOT be the same"?

Reply:
These 3 values are the same if converted to double

Reply:
I also find the text a bit confusing. That _same_ could mean:
*  `largeValue = (long) ((double) largeValue)`, which is what I (and maybe Ankit) understood first.
* `(double) largeValue = (double) (largeValue + 1) = (double) (largeValue + 1)`, which is what I understand you mean.

Suggestion: Let's rephrase that to something like:

> Insert three consecutive numbers that are so large that they are represented by the same double value